src/core/qgsmaplayer.h

```diff
@@ -73,13 +73,17 @@ class CORE_EXPORT QgsMapLayer : public QObject
      */
     QgsMapLayer( QgsMapLayer::LayerType type = VectorLayer, const QString &name = QString::null, const QString &source = QString::null );
 
-    virtual ~QgsMapLayer();
+    /** Copy constructor.
+     * @param mapLayer the map layer to copy
+     * @param deep to copy unique id if true
+     */
+    QgsMapLayer( const QgsMapLayer &mapLayer, bool deep = true );
```

Comment:
Oh! I had not considered this solution. It's definitely better than protected copy destructor...
Thank you @m-kuhn